hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelinePlacementPolicy.java

```diff
@@ -80,7 +81,19 @@ public PipelinePlacementPolicy(
    */
   @VisibleForTesting
   boolean meetCriteria(DatanodeDetails datanodeDetails, long heavyNodeLimit) {
-    return (nodeManager.getPipelinesCount(datanodeDetails) <= heavyNodeLimit);
+    if (heavyNodeLimit == 0) {
+      // no limit applied.
+      return true;
+    }
+    boolean meet = (nodeManager.getPipelinesCount(datanodeDetails)
+        <= heavyNodeLimit);
+    if (!meet) {
+      LOG.info("Pipeline Placement: Doesn't meet the criteria to be viable " +
+          "node： " + datanodeDetails.getUuid().toString() + " Heaviness: " +
```

Comment:
NIT: Doesn't meet the criteria to be viable node => Can't place more pipeline on heavy datanode

Reply:
Sounds good. Updated.